src/app/loadout/LoadoutDrawerContents.tsx

```diff
@@ -205,3 +212,28 @@ function fillLoadoutFromEquipped(
     }
   }
 }
+
+async function fillLoadoutfromInventory(
+  loadout: Loadout,
+  stores: DimStore[],
+  add: (item: DimItem, e?: MouseEvent, equip?: boolean) => void
+) {
+  if (!loadout) {
+    return;
+  }
+  const dimStore =
+    (loadout.classType !== DestinyClass.Unknown &&
+      stores.find((s) => s.classType === loadout.classType)) ||
+    getCurrentStore(stores)!;
+
+  const items = dimStore.items.filter(
+    (item) => itemCanBeInLoadout(item) && fromEquippedTypes.includes(item.type)
+  );
+
+  for (const item of items) {
+    if (item.bucket.sort === 'Armor' || item.bucket.sort === 'Weapons') {
```

Comment:
You can add `fromEquippedTypes.includes(item.type)` to the conditions in your `filter` above, which will match how the `fillLoadoutFromEquipped` chooses which types can be included.